### PR TITLE
Elastic bulk request + single sample version

### PR DESCRIPTION
**Pull request by kevinstyp**
This closes issue #49

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -255,7 +255,7 @@
         <dependency>
             <groupId>org.elasticsearch.client</groupId>
             <artifactId>elasticsearch-rest-high-level-client</artifactId>
-            <version>7.0.1</version>
+            <version>6.6.2</version>
```
Comment (flohannes) [inline pom.xml:260]: is there a newer elasticsearch version, which we can use?

File: src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        }
+        this.client = new RestHighLevelClient(
+                RestClient.builder(httpHosts));
+    }
+
+    private static List<Pair<String, Integer>> convertHostPortArgs(String tags) {
+        return Arrays.stream(tags.split(",")).map(String::trim)
+                .map(s -> {
+                    String[] hostPorts = s.split(":");
+                    return new Pair<>(hostPorts[0], Integer.valueOf(hostPorts[1]));
```
Comment (flohannes) [inline src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java]: do we need to do checks before getting indexes on this array?

File: src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                }).collect(Collectors.toList());
+    }
+
+    public void writeSingle(Sample sample) throws IOException {
```
Comment (flohannes) [inline src/main/java/bitflow4j/steps/database/ElasticsearchUtil.java]: would rename writeSingle and writeBulk both to write

File: src/main/java/bitflow4j/steps/database/WriteHistogramToElasticsearch.java [hunk 4]
```diff
@@ -40,83 +22,17 @@
...
-                RestClient.builder(httpHosts));
-    }
-
-    private static List<Pair<String, Integer>> convertHostPortArgs(String tags) {
-        return Arrays.stream(tags.split(",")).map(String::trim)
-                .map(s -> {
-                    String[] hostPorts = s.split(":");
-                    return new Pair<>(hostPorts[0], Integer.valueOf(hostPorts[1]));
-                }).collect(Collectors.toList());
+        this.elasticsearchUtil = new ElasticsearchUtil(hostPorts, indexName, identifierKey, identifierTemplate);
```
Comment (flohannes) [inline src/main/java/bitflow4j/steps/database/WriteHistogramToElasticsearch.java]: this. not needed here

File: src/main/java/bitflow4j/steps/database/WriteToElasticsearchDB.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    /**
+     * @param hostPorts          Comma-separated list of host:port pairs, e.g.: host1:port1, host2:port2
+     * @param indexName          Index name of the respective index in the Elasticsearch-Database
+     * @param identifierKey      Name of key-tag which is saved for each sample and used to identify/filter the entry in
+     *                           the database (the same name will be applied in the DB entry as the property name)
+     * @param identifierTemplate Used template to fill the named property with meaningful content (Tag-templates
+     *                           should be used here)
+     */
+    public WriteToElasticsearchDB(String hostPorts, String indexName, String identifierKey, String identifierTemplate) {
+        this.elasticsearchUtil = new ElasticsearchUtil(hostPorts, indexName, identifierKey, identifierTemplate);
```
Comment (flohannes) [inline src/main/java/bitflow4j/steps/database/WriteToElasticsearchDB.java]: this. not needed

**Review by flohannes — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (kevinstyp) [reply pom.xml:260]: Yes and no and yes ;):
The currently used grafana version supports only elastic 6.6, but 6 days ago a newly released Grafana version (6.2) was released which supports elastic 7, I do not know how stable this works. Elastic 7 itself already released versions 7.1 and 7.2.

Comment (flohannes) [reply pom.xml:260]: please add comment that the version is due to grafana 6.2

**Review by flohannes — APPROVED** [review]

**Pull request merged**